packages/library/src/rpc-transport.ts

```diff
@@ -30,5 +38,5 @@ export function createDefaultRpcTransport(config: Parameters<typeof createHttpTr
             },
         }),
         transport => getRpcTransportWithRequestCoalescing(transport, getSolanaRpcPayloadDeduplicationKey),
-    );
+    ) as IRpcTransportFromClusterUrl<TClusterUrl>;
```

Comment:
I may be wrong but there's a change you don't need this cast if you pass `TClusterUrl` to the `createHttpTransport` function like so:

```ts
pipe(
    createHttpTransport<TClusterUrl>(...),
    // ...
);
```

Reply:
So, I can make this work if I modify the second function in the `pipe` to return the provided transport as a pass-through type, but it involves a cast in there (ie. `as TTransport`).

However, idiomatically I think it does make more sense for that function to cast than it does for `createHttpTransport`, so I think I'm going to add it in.